web/src/enterprise/campaigns/detail/CampaignDetails.tsx

```diff
@@ -417,22 +458,76 @@ export const CampaignDetails: React.FunctionComponent<Props> = ({
                                         type="button"
                                         className="btn btn-secondary mr-1"
                                         onClick={onEdit}
-                                        disabled={mode === 'deleting'}
+                                        disabled={mode === 'deleting' || mode === 'closing'}
                                     >
                                         Edit
                                     </button>
-                                    <button
-                                        type="button"
-                                        className="btn btn-danger"
-                                        onClick={onDelete}
-                                        disabled={mode === 'deleting'}
-                                    >
-                                        Delete
-                                    </button>
+                                    {campaign && campaign.__typename === 'Campaign' && !campaign.closedAt && (
+                                        <details className="campaign-details__details">
+                                            <summary>
+                                                <span className="btn btn-secondary mr-1 dropdown-toggle">Close</span>
+                                            </summary>
+                                            <div className="position-absolute campaign-details__details-menu">
+                                                <div className="card mt-1">
+                                                    <div className="card-body">
+                                                        <p>
+                                                            <b>You are about to close this campaign</b>
```

Comment:
```suggestion
                                                            <b>Close campaign</b>
```

Reply:
For consistency with our other delete confirmations (see deleting a saved search or external service). You'll need to fix the variable here.
```suggestion
                                                            <b>Delete campaign {campaignName}?</b>
```

Reply:
names can be somewhat long for campaigns 🤔 do you think it'll cause confusion of we truncate the name?

in the case where I have 2 campaigns named
- Remove all npm packages calle...[d "lodash*"]
- Remove all npm packages calle...[d "puppeteer"]

Reply:
hmmmm good point. Does it look really bad if we let the line wrap with the title (if needed) rather than truncating?

Reply:
it actually looks good, let's go with that